crates/bevy_reflect/src/impls/std.rs

```diff
@@ -1054,44 +1204,41 @@ impl Reflect for &'static Path {
         <Self as Typed>::type_info()
     }
 
-    fn into_any(self: Box<Self>) -> Box<dyn Any> {
-        self
+    fn as_full(&self) -> Option<&dyn Reflect> {
+        Some(self)
     }
 
-    fn as_any(&self) -> &dyn Any {
-        self
+    fn as_full_mut(&mut self) -> Option<&mut dyn Reflect> {
+        Some(self)
     }
 
-    fn as_any_mut(&mut self) -> &mut dyn Any {
-        self
+    fn into_full(self: Box<Self>) -> Result<Box<dyn Reflect>, Box<dyn PartialReflect>> {
+        Ok(self)
     }
 
-    fn into_reflect(self: Box<Self>) -> Box<dyn Reflect> {
+    fn as_partial(&self) -> &dyn PartialReflect {
         self
     }
 
-    fn as_reflect(&self) -> &dyn Reflect {
+    fn as_partial_mut(&mut self) -> &mut dyn PartialReflect {
         self
     }
 
-    fn as_reflect_mut(&mut self) -> &mut dyn Reflect {
+    fn into_partial(self: Box<Self>) -> Box<dyn PartialReflect> {
         self
     }
 
-    fn apply(&mut self, value: &dyn Reflect) {
-        let value = value.as_any();
-        if let Some(&value) = value.downcast_ref::<Self>() {
+    fn apply(&mut self, value: &dyn PartialReflect) {
+        if let Some(&value) = value
+            .as_full()
+            .and_then(<dyn Reflect>::downcast_ref::<Self>)
```

Comment:
Might as well consider using `try_downcast_ref` here instead of casting to `Reflect` first, as you do in many different places.

Reply:
This wasn't addressed. Wouldn't `try_downcast_ref` make the code a bit cleaner?